utils/stack_dump.go

```diff
@@ -0,0 +1,17 @@
+package utils
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+var DebugDataDir = ""
+
+func FullGoroutineStackDump() error {
+	buffer := make([]byte, 128*1024*1024)
+	length := runtime.Stack(buffer, true)
+
+	fileName := filepath.Join(DebugDataDir, "full-goroutine-stack-dump.txt")
+	return os.WriteFile(fileName, buffer[:length], 0600)
+}
```

Comment:
@moshe-blox this not used here and not related to this PR, do we still want it here for some reason?